### DISCUSSION TITLE
Too long names for packaging: is this still a valid restriction?

### DISCUSSION THREAD

**Discussion by rubyFeedback** [ruby/rubygems#5645] [Ideas]
Today I kind of created a new file for my gem called "studium".

When trying to package it I got this error:

    ERROR:  While executing gem ... (Gem::Package::TooLongFileName)
        File         

"lib/studium/yaml/curricula/outdated/bachelor_bioinformatik_und_molekulare_biotechnologie_unter_berücksichtigung_medizinisch_immunologischer_aspekte.yml"

    has a too long name (should be 100 or less) 
    # oddly enough I can not even indent this via 4 ' ' here on github hehe ...
    # when I do it simply makes a newline instead...

I guess the file path also matters. But is this restriction still effective? If I remember correctly
this has been in place due to zlib from years ago? Perhaps that restriction could be dropped, or 
alternatively, the threshold raised to 120 or something. Right now I move the 
offending files away, build the .gem, upload it, then put the .yml files back in 
place, but this is a bit tedious. This is why I'd love to see this restriction be dropped or
the threshold be lessened (not even sure why it is at 100? Is there a reason for
this or is that a magic number? 99? 101?).

**Comment by deivid-rodriguez** [comment]
I don't know. Feel free to investigate.

**Comment by deivid-rodriguez** [comment]
In all these years, nobody has ever needed or wanted to ship a file with a name 100 characters long inside a gem, so I doubt this will ever be implemented (if even possible) unless you implement it yourself. I'm going to move this to discussion since we're not even sure it's a valid (acceptable) feature request.

**Comment by duckinator** [comment]
The 100 character limitation is inherited from the UStar ("Unix Standard TAR") standard, which is the newest standardized tar format.

Each file or directory is stored in a tar file as a _record_, and the name of each record can only be 100 characters long. See https://pubs.opengroup.org/onlinepubs/9699919799/utilities/pax.html#tag_20_92_13_06 for details.
